plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/clients/admin/SslKafkaAdminFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.kafka.clients.admin;
+
+import io.trino.plugin.kafka.security.ForKafkaSsl;
+import io.trino.plugin.kafka.security.KafkaSslConfig;
+import io.trino.spi.connector.ConnectorSession;
+
+import javax.inject.Inject;
+
+import java.util.Properties;
+
+import static java.util.Objects.requireNonNull;
+
+public class SslKafkaAdminFactory
+        implements KafkaAdminFactory
+{
+    private final Properties securityProperties;
```

Comment:
It would be better to keep it as `Map` (Immutable). `Properties` is mutable object, so one could question if it is safe.

Reply:
Hi @kokosing , by immutable `Map`, are you referencing to `ImmutableMap` from Guava library?

Reply:
Yes.

Reply:
Thank you for the thorough review!